File: examples/js/presigned-urls.js
import {
  S3Client,
  GetObjectCommand,
  PutObjectCommand,
  DeleteObjectCommand,
} from "@aws-sdk/client-s3";
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";

const S3 = new S3Client({
  region: "auto",
  s3ForcePathStyle: false,
});

// Presigned GET, allows users to download objects without making the bucket public.
console.log(
  "GET:",
  await getSignedUrl(
    S3,
    new GetObjectCommand({ Bucket: "tigris-example", Key: "bar.txt" }),
    { expiresIn: 3600 }, // 1 hour
  ),
);

// Presigned PUT, allows users to upload objects without going through your server.
console.log(
  "PUT:",
  await getSignedUrl(
    S3,
    new PutObjectCommand({ Bucket: "tigris-example", Key: "bar.txt" }),
    { expiresIn: 3600 }, // 1 hour
  ),
);

// Presigned DELETE, allows users to delete objects.
console.log(
  "DELETE:",
  await getSignedUrl(
    S3,
    new DeleteObjectCommand({ Bucket: "tigris-example", Key: "bar.txt" }),
    { expiresIn: 3600 }, // 1 hour
  ),
);

File: examples/ruby/presigned_urls.rb
require "aws-sdk-s3"

s3 = Aws::S3::Client.new(
    region: "auto",
    endpoint: "https://t3.storageapi.dev",
    force_path_style: false,
)

bucket_name = "tigris-example"
bucket = Aws::S3::Bucket.new(name: bucket_name, client: s3)

presigned_url = bucket.object("test-object").presigned_url(:get, expires_in: 3600)
puts "Presigned URL for GET: #{presigned_url}"